cheatsheets/Docker_Security_Cheat_Sheet.md

```diff
@@ -0,0 +1,133 @@
+```
+
+**And remeber: Do not run containers with the *--privileged* flag!!!**
+
+## RULE \#4 - Add –no-new-privileges flag
+
+Always run your docker images with `--security-opt=no-new-privileges` in order to prevent escalate privileges using `setuid` or `setgid` binaries.
+
+## RULE \#5 - Disable inter-container communication (--icc=false)
+
+By default inter-container communication (icc) is enabled - it means that all containers can talk with each other (using `docker0` bridged network). 
+This can be disabled by running docker deamon with `--icc=false` flag. 
+If icc is disabled (icc=false) it is required to tell which containers can communicate using --link=CONTAINER_NAME_or_ID:ALIAS option. 
+See more in [Docker documentation - container communication](https://docs.docker.com/v17.09/engine/userguide/networking/default_network/container-communication/#communication-between-containers)
+
+## RULE \#6 - Use Linux Security Module (seccomp, AppArmor, or SELinux)
+
+**First of all do not disable default security profile!** 
+
+Consider using security profile like [seccomp](https://docs.docker.com/engine/security/seccomp/) or [AppArmor](https://docs.docker.com/engine/security/apparmor/). 
+
+## RULE \#7 - Limit resources (memory, CPU, file descriptors, processes, restarts)
+
+The best way to avoid DoS attacks is limiting resources. You can limit [memory](https://docs.docker.com/config/containers/resource_constraints/#memory), [CPU](https://docs.docker.com/config/containers/resource_constraints/#cpu), maximum number of restarts (`--restart=on-failure:<number_of_restarts>`), maximum number of file descriptors (`--ulimit nofile=<number>`) and maximum number of processes (`--ulimit nproc=<number>`).
+
+[Check documentation for more details about ulimits](https://docs.docker.com/engine/reference/commandline/run/#set-ulimits-in-container---ulimit)
+
+## RULE \#8 - Set filesystem and volumes to read-only 
+
+**Run containers with a read-only filesystem** using `--read-only` flag. For example:
+
+```
+docker run --read-only alpine sh -c 'echo "whatever" > /tmp'
+```
+
+If application inside container have to save something temporarily combine `--read-only` flag with `--tmpfs` like this:
+
+```
+docker run --read-only --tmpfs /tmp alpine sh -c 'echo "whatever" > /tmp/file'
+```
+
+Equivalent in docker-compose file will be:
+
+```
+version: "3"
+services:
+  alpine:
+    image: alpine
+    read_only: true
+
+```
+
+In addition if volume is mounted only for reading **mount them as a read-only**
+It can be done by appending `:ro` to the `-v` like this:
+
+```
+docker run -v volume-name:/path/in/container:ro alpine
+```
+
+Or by using `--mount` option:
+
+```
+$ docker run --mount source=volume-name,destination=/path/in/container,readonly alpine
+```
+
+## RULE \#9 - Use static analysis tools
+
+Scan images using open source tools like [anchore](https://anchore.com/) or [Clair](https://github.com/coreos/clair) to detect containers with known vulnerabilities. 
```

Comment:
Can I suggest that we also add [Snyk](https://snyk.io) as an open source and free tool (even for private projects) to scan and monitor for known vulnerabilities in docker images?
Snyk also offers remediation advice and suggests alternate images with less vulnerabilities to update to.

disclaimer: I'm a developer advocate at Snyk ;-)

Reply:
Split the analysis tools reco links between 2 groups:
* Free and open source ones.
* Commercial ones.

For commercial add at least 2 differents vendors because we must stay vendor neutral. If we add Snyk then we must add another vendor like add [JFrog XRay](https://jfrog.com/xray/) or [Qualys](https://www.qualys.com/apps/container-security/) or [AquaSec](https://www.aquasec.com/products/open-source-projects/)...